examples/convecting_plankton.jl

```diff
@@ -1,88 +1,135 @@
-# # Convecting plankton
+# # Plankton mixing and blooming
 #
-# In this example, two-dimensional convection into a stratified fluid
-# mixes a phytoplankton-like tracer. This example demonstrates:
+# In this example, we simulate the mixing of phytoplankton by convection
+# that decreases in time and eventually shuts off, thereby precipitating a
+# phytoplankton bloom. A similar scenario was simulated by
+# [Taylor and Ferrari (2011)](https://aslopubs.onlinelibrary.wiley.com/doi/abs/10.4319/lo.2011.56.6.2293),
+# providing evidence that the
+# ["critical turbulence hypothesis"](https://en.wikipedia.org/wiki/Critical_depth#Critical_Turbulence_Hypothesis)
+# explains the explosive bloom of oceanic phytoplankton
+# observed in spring.
 #
-#   * How to set boundary conditions.
-#   * How to insert a user-defined forcing function into a model.
+# The phytoplankton in our model advect, diffuse, grow, and die according to
+#
+# ```math
+# ∂_t P + \bm{u} ⋅ ∇P - κ ∇²P = (μ₀ \exp(z / λ) - m) \, P \, ,
+# ```
+#
+# where ``\bm{u}`` is the turbulent velocity field, ``κ`` is an isotropic diffusivity,
+#  ``μ₀`` is the phytoplankton growth rate at the surface, ``λ`` is the scale over
+# which sunlight attenuates away from the surface, and ``m`` is the mortality rate
+# of phytoplankton due to viruses and grazing by zooplankton. We use Oceananigans'
+#  `Forcing` abstraction to implement the phytoplankton dynamics described by the right
+# side of the phytoplankton equation above.
+#
+# This example demonstrates
+#
+#   * How to use a user-defined forcing function to
+#     simulate the dynamics of phytoplankton growth in sunlight
+#     and grazing by zooplankton.
+#   * How to set time-dependent boundary conditions.
 #   * How to use the `TimeStepWizard` to adapt the simulation time-step.
 #   * How to use `AveragedField` to diagnose spatial averages of model fields.
 #
 # ## The grid
 #
-# We use a two-dimensional grid with 128² points and 2 m grid spacing:
+# We use a two-dimensional grid with 64² points and 1 m grid spacing:
 
 using Oceananigans
 
-grid = RegularCartesianGrid(size=(128, 1, 128), extent=(64, 1, 64))
+grid = RegularCartesianGrid(size=(64, 1, 64), extent=(64, 1, 64))
 
 # ## Boundary conditions
 #
-# We impose buoyancy loss at the surface with the buoyancy flux
+# We impose a time-dependent, decaying buoyancy loss at the surface,
 
-Qb = 1e-8 # m³ s⁻²
-nothing # hide
+using Oceananigans.Utils
+
+buoyancy_flux(x, y, t, p) = p.initial_buoyancy_flux * exp(-t^4 / (24 * p.shut_off_time^4))
 
-# Note that a _positive_ flux at the _top_ boundary means that buoyancy is
-# carried _upwards_, out of the fluid. This reduces the fluid's buoyancy 
-# near the surface, causing convection.
+buoyancy_flux_parameters = (initial_buoyancy_flux = 1e-8, # m² s⁻³
+                                    shut_off_time = 2hours)
+
+buoyancy_flux_bc = BoundaryCondition(Flux, buoyancy_flux, parameters = buoyancy_flux_parameters)
+
```

Comment:
add `nothing #hide` here

Reply:
I wanted to show it, since we have nice show methods for `BoundaryCondition`

Reply:
oh, if it was intentional then I rest my case :)